test/integration/targets/docker_swarm_facts/tasks/test_swarm_facts.yml

```diff
@@ -0,0 +1,139 @@
+---
+- block:
+  - name: Make sure we're not already using Docker swarm
+    docker_swarm:
+      state: absent
+      force: true
+
+  - name: Try to get docker_swarm_facts when docker is not running in swarm mode
+    docker_swarm_facts:
+    ignore_errors: yes
+    register: output
+
+  - name: assert failure when called when swarm is not in use or not run on mamager node
+    assert:
+      that:
+         - 'output.failed'
+         - 'output.msg == "Error running docker swarm module: must run on swarm manager node"'
+         - 'output.can_talk_to_docker'
```

Comment:
You should explicitly check for `true` (otherwise, it suffices if it is truish).